### ISSUE TITLE
Warn if code violates Law of Demeter

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#362]
If the code consists of "train wrecks", we should report a warning because that violates the [Law of Demeter (LoD)](https://en.wikipedia.org/wiki/Law_of_Demeter).
The intention is to write code that is "shy" and follows the "information hiding" principle, to minimize maintenance efforts.

E.g. following method violates the LoD as it knows too much and is brittle to changes.
```c#
public string DoSomething(A a)
{
    return a.B.C.D.E.F.G.H.GetSomeText();
}
```
